Implement modernized contest list layout and UI updates

rahmanfaizur opened the pull request:
# Modernize Contest List UI

This PR updates the Contest List interface with a more modern, responsive, and user-friendly layout.

## ✨ Key Changes
- Replaced the old tabbed layout with **horizontally scrollable sections** for Current, Future, and Past contests.
- Added a **“View All” grid page** with responsive 3/2/1-column cards.
- Added **300ms search debounce** for smoother performance.
- Updated `ContestCard` styling and improved spacing/visual clarity.
- Improved store logic for resetting lists during search/filter updates.

## 📸 Screenshots

<img width="1920" height="1080" alt="Screenshot from 2025-11-29 02-40-27" src="https://github.com/user-attachments/assets/2a1b0454-3144-4f0e-9027-ea07ea5be312" />
<img width="1920" height="1080" alt="Screenshot from 2025-11-29 02-40-33" src="https://github.com/user-attachments/assets/49dd140c-6f70-4623-9707-a0e8e06bdbb3" />
<img width="1920" height="1080" alt="Screenshot from 2025-11-29 02-40-41" src="https://github.com/user-attachments/assets/772c7310-275e-4d93-87a7-2ddc641ea310" />


## ✔️ Checklist

- [x] Code follows the [omegaUp Coding Guidelines](https://github.com/omegaup/omegaup/blob/main/frontend/www/docs/Coding-guidelines.md)
- [x] Linting and type checks pass
- [x] Manual testing completed (navigation, search, scrolling, grid view)
---
Fixes: #8536

pabo99 reviewed (commented):
Ok, it looks like the test failures are legitimate.

Let’s do this: we need to split this PR into smaller parts because reviewing everything together will take too much time. My suggestion is to move all backend-related changes into a new PR. Since you’ll already be moving them, we can use that new PR to fix the `PHPUnit` failures:

```bash
There were 2 failures:

1) ContestListTest::testContestListOrder with data set #0 ('title', array(4, 3, 2, 1), false)
Failed asserting that two strings are identical.

-'contest_4'
+'contest_1'

/home/runner/work/omegaup/omegaup/frontend/tests/controllers/ContestListTest.php:973

2) ContestListTest::testContestListOrder with data set #3 ('organizer', array(4, 1, 2, 3), false)
Failed asserting that two strings are identical.

-'contest_4'
+'contest_3'

/home/runner/work/omegaup/omegaup/frontend/tests/controllers/ContestListTest.php:973
```

You can run the test individually inside the docker container with:

```bash
$ docker exec -it omegaup-frontend-1 /bin/bash

$ ./stuff/run-php-tests.sh frontend/tests/controllers/ContestListTest.php
```
If you fix the test, push the update in that same PR.

Then we’ll continue splitting the rest to make reviews faster.

rahmanfaizur commented:
Thanks for the guidance. I have followed your suggestion:

Backend: I moved the backend changes to a new PR and fixed the ContestListTest failures there : https://github.com/omegaup/omegaup/pull/8564
Frontend: I updated this PR to contain only the frontend changes.
